### PR TITLE
Add new source + Tweak

### PR DESCRIPTION
**Pull request by dragonx943**
Add SayHentai (https://github.com/KotatsuApp/kotatsu-parsers/issues/1172), tweak DocTruyen5s and update sources domain

### CODE REVIEW

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/vi/DocTruyen5s.kt [hunk 1]
```diff
@@ -1,12 +1,50 @@
...
 internal class DocTruyen5s(context: MangaLoaderContext) :
-	LilianaParser(context, MangaParserSource.DOCTRUYEN5S, "dongmoe.com")
+	LilianaParser(context, MangaParserSource.DOCTRUYEN5S, "dongmoe.com") {
+
+    override suspend fun getPages(chapter: MangaChapter): List<MangaPage> {
+        val fullUrl = chapter.url.toAbsoluteUrl(domain)
+        val doc = webClient.httpGet(fullUrl).parseHtml()
+
+        val script = doc.selectFirst("script:containsData(const CHAPTER_ID)")?.data()
+            ?: throw Exception("Failed to get chapter id")
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/vi/DocTruyen5s.kt:21]: Do not use generic `Exception` class

Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/vi/DocTruyen5s.kt:21]:
```suggestion
        val script = doc.selectFirstOrThrow("script:containsData(const CHAPTER_ID)")?.data()
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/vi/DocTruyen5s.kt [hunk 2]
```diff
@@ -1,12 +1,50 @@
...
+            append("https://")
+            append(domain)
+            append("/ajax/image/list/chap/")
+            append(chapterId)
+        }
+
+        val responseJson = webClient.httpGet(ajaxUrl).parseJson()
+
+        if (!responseJson.optBoolean("status", false)) {
+            throw Exception(responseJson.optString("msg"))
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/vi/DocTruyen5s.kt:35]:
```suggestion
            throw IllegalStateException(responseJson.optString("msg"))
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/SayHentai.kt [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            )
+        }
+    }
+
+    override suspend fun getDetails(manga: Manga): Manga {
+        val doc = webClient.httpGet(manga.url.toAbsoluteUrl(domain)).parseHtml()
+        return manga.copy(
+            altTitle = doc.selectFirst("h2.other-name")?.text(),
+            author = doc.selectFirst("div.summary-heading:contains(Tác giả) + div.summary-content")?.text(),
+            tags = doc.select("div.genres-content a[rel=tag]").mapNotNullToSet { a ->
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/SayHentai.kt:96]:
```suggestion
            tags = doc.select("div.genres-content a[rel=tag]").mapToSet { a ->
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/SayHentai.kt [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+                MangaTag(
+                    key = a.attr("href").substringAfterLast("/"),
+                    title = a.text(),
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/SayHentai.kt:99]:
```suggestion
                    title = a.text().toTitleCase(sourceLocale),
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/SayHentai.kt [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    override suspend fun getDetails(manga: Manga): Manga {
+        val doc = webClient.httpGet(manga.url.toAbsoluteUrl(domain)).parseHtml()
+        return manga.copy(
+            altTitle = doc.selectFirst("h2.other-name")?.text(),
+            author = doc.selectFirst("div.summary-heading:contains(Tác giả) + div.summary-content")?.text(),
+            tags = doc.select("div.genres-content a[rel=tag]").mapNotNullToSet { a ->
+                MangaTag(
+                    key = a.attr("href").substringAfterLast("/"),
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/SayHentai.kt:98]:
```suggestion
                    key = a.attr("href").substringAfterLast('/'),
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/SayHentai.kt [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+            SortOrder.RATING -> "m_orderby=rating"
+            else -> "m_orderby=latest"
+        }
+    }
+
+    private suspend fun fetchTags(): Set<MangaTag> {
+        return tagsCache.get() ?: run {
+            val tags = webClient.httpGet("https://$domain/genre").parseHtml()
+                .select("ul.page-genres li a")
+                .mapNotNullToSet { a ->
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/SayHentai.kt:169]:
```suggestion
                .mapToSet { a ->
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/SayHentai.kt [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+                    val title = a.ownText().trim()
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/vi/SayHentai.kt:170]:
```suggestion
                    val title = a.ownText().toTitleCase(sourceLocale)
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/vi/DocTruyen5s.kt (hunk 1 above)
Comment (dragonx943) [reply src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/vi/DocTruyen5s.kt:21]: throw e ?

**Pull request merged**